build_system/seqan3-config.cmake

```diff
@@ -599,6 +599,15 @@ if (NOT ${FIND_NAME}_FIND_QUIETLY)
     message (STATUS "${ColourBold}Found SeqAn3:${ColourReset} ${SEQAN3_BASEDIR}/seqan3 (found version \"${SEQAN3_VERSION_STRING}\")")
 endif ()
 
+separate_arguments (SEQAN3_CXX_FLAGS_LIST UNIX_COMMAND "${SEQAN3_CXX_FLAGS}")
+string(REGEX REPLACE ";( )*[-/]D" ";" SEQAN3_DEFINITIONS_LIST ";${SEQAN3_DEFINITIONS};")
```

Comment:
this splits on other special characters, as well?

Do wo know that the above work work for all the flags possibly set by seqan3-config?

Reply:
Yes, since DEFINITIONS should only define `-DMY_DEF` `/DMY_DEF`. This is inspired by `FindMPI`. If you want to add compiler flags you should use `target_compile_options`.

If we would require cmake 3.11 we could use [target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html) directly. Before, it was not allowed to use the `target_*` functions to add properties to a `INTERFACE IMPORTED`